### PR TITLE
feat(ui5-step-input): add value-state-change event

### PR DESCRIPTION
**Pull request by tsanislavgatev**
fixes: https://github.com/SAP/ui5-webcomponents/issues/4791
fixes: https://github.com/SAP/ui5-webcomponents/issues/5130

### CODE REVIEW

**Review by hinzzx — APPROVED** [review]

File: packages/main/src/StepInput.ts [hunk 1]
```diff
@@ -106,6 +111,32 @@ const INITIAL_SPEED = 120; // milliseconds
...
 @event("change")
+/**
+ * Fired before the value state of the component is updated internally.
+ * The event is preventable, meaning that if it's default action is
+ * prevented, the component will not update the value state.
+ *
+ * @allowPreventDefault
+ * @public
+ * @param {string} valueState The new <code>valueState</code> that will be set.
+ * @param {boolean} valid Indicator if the value is in correct format pattern and in valid range.
```
Comment (hinzzx) [inline packages/main/src/StepInput.ts:122]: The documentation here says format-pattern.

Maybe we can change it to - Indicator if the value is correct and in range (if limitations are given with <code>min</code> or <code>max</code> properties.

**Review by hinzzx — CHANGES_REQUESTED** [review]

File: packages/main/src/StepInput.ts (hunk 1 above)
Comment (tsanislavgatev) [reply packages/main/src/StepInput.ts]: that's why we should not copy and paste documentation :D . Fixed.

**Review by hinzzx — APPROVED** [review]

File: packages/main/src/StepInput.ts [hunk 2]
```diff
@@ -106,6 +111,32 @@ const INITIAL_SPEED = 120; // milliseconds
  * @public
  */
 @event("change")
+/**
+ * Fired before the value state of the component is updated internally.
+ * The event is preventable, meaning that if it's default action is
+ * prevented, the component will not update the value state.
+ *
+ * @allowPreventDefault
+ * @public
```
Comment (nnaydenow) [inline packages/main/src/StepInput.ts:120]: Add since tag

**Review by nnaydenow — CHANGES_REQUESTED** [review]

File: packages/main/test/specs/StepInput.spec.js [hunk 3]
```diff
@@ -456,6 +456,16 @@ describe("'change' event firing", () => {
 		assert.strictEqual(Number(await changeResult.getProperty("value")), 2, "'change' event is fired 2 times");
 	});
 
+	it("Value state is not changed, when value-state-change is prevented", async () => {
+		const input = await browser.$("#stepInputValueStateChange").shadow$("ui5-input").shadow$("input");
+
+		const valueState = await input.getProperty("valueState");
+		await input.click();
```
Comment (nnaydenow) [inline packages/main/test/specs/StepInput.spec.js:463]: this test doesn't test the functionality. Basically validation is done when the focus leaves the input and here the focus remains in the input even after change.

I would suggest you to check also whether the event was fire, not only if it was prevented.

**Review by nnaydenow — CHANGES_REQUESTED** [review]

**Review by nnaydenow — APPROVED** [review]

**Pull request merged**